sources/packages/backend/apps/queue-consumers/src/processors/schedulers/base-queue.ts

```diff
@@ -0,0 +1,74 @@
+import { CustomNamedError, parseJSONError } from "@sims/utilities";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import {
+  PROCESS_SUMMARY_CONTAINS_ERROR,
+  logProcessSummaryToJobLogger,
+} from "../../utilities";
+import { Job } from "bull";
+import { Process } from "@nestjs/bull";
+
+/**
+ * Provides basic functionality for queue processing.
+ */
+export abstract class BaseQueue<T> {
+  /**
+   * Wrap the queue job execution in a try/catch for the global error handling.
+   * It provides also the default {@link ProcessSummary} for logging.
+   * @param job process job.
+   * @returns processing result.
+   */
+  @Process()
+  async processQueue(job: Job<T>): Promise<string | string[]> {
+    const processSummary = new ProcessSummary();
+    try {
+      this.logger.log(`Processing queue ${job.queue.name}, job ID ${job.id}.`);
+      const result = await this.process(job, processSummary);
+      processSummary.info(`${job.queue.name}, job ID ${job.id}, executed.`);
+      const logsSum = processSummary.getLogLevelSum();
+      if (logsSum.error) {
```

Comment:
With logSum errors now certainly resulting in retry(being thrown) we should have a clear understanding with the team about when to use warning and error messages.